### ISSUE TITLE
Deployment maps require using OUId as target to resolve OU specific parameters

### ISSUE THREAD

**Issue by hynynen** [awslabs/aws-deployment-framework#294]
I have this deployment map:
```
ipelines:
  - name: vpc-pipeline
    default_providers:
      source:
        provider: codecommit
        properties:
          account_id: 11223344556677
      deploy:
        provider: cloudformation
        properties:
            action: replace_on_failure
            stack_name: ipam-vpc
            template_filename: packaged.yml
    params:  
      restart_execution_on_update: True
    targets:
      - target: ou-tio5-4fqvuabz # OU: /foo/bar
        regions: [eu-west-1, eu-north-1]
```
And the vpc repo has parameter files:
```
params/
├── global.yml
├── ou-123-445678.yml
└── ou-123-567890.yml
```
This works fine. If I change target syntax in deployment map to use human-readable names of OUs (e.g. /foo/bar), generating parameters fails.

It is mentioned in documentation that "When using OU parameter files, the OU must be specified in the deployment map as a target", but this behavior is not clear nor convenient. ADF should be able to deduce OU id from OU name in an organization.

**Comment by niklaswesterstrahleknowit** [comment]
I would also have a related request, we're using tags as target - but would have need to have parameters based on OU. 

So I'd actually request ou based parameter files to be processed, regardless of target being the ou or anything else.

**Comment by sbkok** [comment]
Hi @hynynen, with PR #559, the `generate_params.py` script was refactored. Improving a number of issues that we ran into, including yours.

With the updated implementation, if you target the account with an OU path (`/foo/bar`), it will search for the parameter files at:
* params/foo/bar_${region}.json
* params/foo/bar_${region}.yml
* params/foo/bar.json
* params/foo/bar.yml

Please note: any character other than alphanumeric, dash, underscore, and slash characters are replaced underscores. This improvement is scheduled to be released as part of ADF v3.2.0, which will be released shortly.

**Comment by sbkok** [comment]
@niklaswesterstrahleknowit your request is unfortunately not supported at this moment. The context information to what OU an account belongs is only added when the OU was used as a selection mechanism to match the account.

Could you please open a separate issue to request this change? As the issue you added it too will get closed with the next release. And your suggestion sounds like a valuable improvement. I don't want to lose track of that.

**Comment by sbkok** [comment]
Thank you for your patience. I am happy to inform you that this feature has been [released as part of v3.2.0](https://github.com/awslabs/aws-deployment-framework/releases/tag/v3.2.0) just now.
I'm hereby closing this issue. Please open a new issue if you are experiencing any issues related to this feature.
